src/render/ngage/SDL_render_ngage.cpp

```diff
@@ -0,0 +1,790 @@
+        {
+            aPoints[i] = TPoint(aVerts[i].x, aVerts[i].y);
+        }
+
+        TUint32 aColor = (
+            ((TUint8)aVerts->color.a << 24) |
+            ((TUint8)aVerts->color.b << 16) |
+            ((TUint8)aVerts->color.g << 8) |
+            (TUint8)aVerts->color.r);
+
+        iRenderer->Gc()->SetPenColor(aColor);
+        iRenderer->Gc()->DrawPolyLineNoEndPoint(aPoints, aCount);
+
+        delete[] aPoints;
+    }
+}
+
+void CRenderer::DrawPoints(NGAGE_Vertex* aVerts, const TInt aCount)
+{
+    if (iRenderer && iRenderer->Gc())
+    {
+        for (TInt i = 0; i < aCount; i++, aVerts++)
+        {
+            TUint32 aColor = (
+                ((TUint8)aVerts->color.a << 24) |
+                ((TUint8)aVerts->color.b << 16) |
+                ((TUint8)aVerts->color.g << 8) |
+                (TUint8)aVerts->color.r);
+
+            iRenderer->Gc()->SetPenColor(aColor);
+            iRenderer->Gc()->Plot(TPoint(aVerts->x, aVerts->y));
+        }
+    }
+}
+
+void CRenderer::FillRects(NGAGE_Vertex* aVerts, const TInt aCount)
+{
+    if (iRenderer && iRenderer->Gc())
+    {
+        for (TInt i = 0; i < aCount; i++, aVerts++)
+        {
+            TPoint pos(aVerts[i].x, aVerts[i].y);
+            TSize size(
+                aVerts[i + 1].x,
+                aVerts[i + 1].y);
+            TRect rect(pos, size);
+
+            TUint32 aColor = (
+                ((TUint8)aVerts->color.a << 24) |
+                ((TUint8)aVerts->color.b << 16) |
+                ((TUint8)aVerts->color.g << 8) |
+                (TUint8)aVerts->color.r);
+
+            iRenderer->Gc()->SetPenColor(aColor);
+            iRenderer->Gc()->SetBrushColor(aColor);
+            iRenderer->Gc()->DrawRect(rect);
+        }
+    }
+}
+
+void CRenderer::Flip()
+{
+    if (!iRenderer)
+    {
+        SDL_Log("iRenderer is NULL.");
+        return;
+    }
+
+    if (!iIsFocused)
+    {
+        return;
+    }
+
+    iRenderer->Gc()->UseFont(iFont);
+
+    if (iShowFPS && iRenderer->Gc())
+    {
+        UpdateFPS();
+
+        TBuf<64> info;
+
+        iRenderer->Gc()->SetPenStyle(CGraphicsContext::ESolidPen);
+        iRenderer->Gc()->SetBrushStyle(CGraphicsContext::ENullBrush);
+        iRenderer->Gc()->SetPenColor(KRgbCyan);
+
+        TRect aTextRect(TPoint(3, 203 - iFont->HeightInPixels()), TSize(45, iFont->HeightInPixels() + 2));
+        iRenderer->Gc()->SetBrushStyle(CGraphicsContext::ESolidBrush);
+        iRenderer->Gc()->SetBrushColor(KRgbBlack);
+        iRenderer->Gc()->DrawRect(aTextRect);
+
+        // Draw messages.
+        info.Format(_L("FPS: %d"), iFPS);
+        iRenderer->Gc()->DrawText(info, TPoint(5, 203));
+    }
+    else
+    {
+        // This is a workaround that helps regulating the FPS.
+        iRenderer->Gc()->DrawText(_L(""), TPoint(0, 0));
+    }
+    iRenderer->Gc()->DiscardFont();
+    iRenderer->Flip(iDirectScreen);
+
+    // Keep the backlight on.
+    User::ResetInactivityTime();
```

Comment:
This looks like a good thing to control via `SDL_EnableScreenSaver`/`SDL_DisableScreenSaver`.

Reply:
I look into it, thanks!

Reply:
`SDL_EnableScreenSaver()` can now be called to allow the backlight to turn off.

Done.